### PR TITLE
Implement TKOFF_THR_IDLE

### PR DESCRIPTION
**Pull request by Georacer**
I thought I'd tackle https://github.com/ArduPilot/ardupilot/issues/20754.

A new parameter `TKOFF_THR_IDLE` has been added and a new option bit 1 to `TKOFF_OPTIONS` to enable it (off by default).

In TAKEOFF and AUTO modes, while in `FlightStage::TAKEOFF` and while the throttle is suppressed, `TKOFF_THR_IDLE` is output to the throttle.

Here's how it looks like in a catapult takeoff, for `TKOFF_THR_IDLE=20`:
image

### CODE REVIEW

File: ArduPlane/Parameters.cpp [hunk 1]
```diff
@@ -158,10 +158,20 @@ const AP_Param::Info Plane::var_info[] = {
...
+    // @Range: 0 100
+    // @Increment: 1
+    // @User: Standard
+    ASCALAR(takeoff_throttle_idle,       "TKOFF_THR_IDLE",    0),
+
     // @Param: TKOFF_OPTIONS
     // @DisplayName: Takeoff options
     // @Description: This selects the mode of the takeoff in AUTO and TAKEOFF flight modes. 
     // @Bitmask: 0: When unset the maximum allowed throttle is always used (THR_MAX or TKOFF_THR_MAX) during takeoff. When set TECS is allowed to operate between a minimum (THR_MIN or TKOFF_THR_MIN) and a maximum (THR_MAX or TKOFF_THR_MAX) limit. Applicable only when using an airspeed sensor.
+    // @Bitmask: 1: When set the throttle will run at TKOFF_THR_IDLE before takeoff.
```
Comment (tridge) [inline ArduPlane/Parameters.cpp:174]: don't need the option bit, just use when non-zero

**Review by tridge — CHANGES_REQUESTED** [review]

File: ArduPlane/servos.cpp [hunk 2]
```diff
@@ -619,6 +619,11 @@ void Plane::set_throttle(void)
             // throttle is suppressed (above) to zero in final flare in auto mode, but we allow instead thr_min if user prefers, eg turbines:
             SRV_Channels::set_output_scaled(SRV_Channel::k_throttle, aparm.throttle_min.get());
 
+        } else if ((flight_stage == AP_FixedWing::FlightStage::TAKEOFF)
+                    && !(aparm.takeoff_throttle_idle == 0)
```
Comment (tridge) [inline ArduPlane/servos.cpp:623]: is_positive(aparm.takeoff_throttle_idle.get())

**Review by tridge — CHANGES_REQUESTED** [review]

**Comment by Georacer** [comment]
Closes #20754 .

**Review by tridge — APPROVED** [review]

**Comment by rtm7777** [comment]
I see this PR in 4.7.0 release notes, did you test it on a plane? I've tried to implement something similar and I had to change much more things and have no time to finish and properly test

**Comment by Georacer** [comment]
It has been tested in simulation. I'm not aware of real-life testing.
We discovered last week that it works only on TAKEOFF mode, not on AUTO, and [the fix](https://github.com/ArduPilot/ardupilot/pull/33784) will likely make its way into 4.7.1.

**Pull request merged**